writer.go

```diff
@@ -18,26 +18,38 @@ type Writer struct {
 	index  *pinecone.IndexConnection
 }
 
-// NewWriter New creates new instance of the Writer.
 func NewWriter(ctx context.Context, config DestinationConfig) (*Writer, error) {
-	sdk.Logger(ctx).Trace().Msg("Creating new writer.")
-
-	sdk.Logger(ctx).Error().Msgf("API: %v, INDEX:%v", config.PineconeAPIKey, config.PineconeHostURL)
-
-	pineconeClient, indexConnection, err := NewPineconeClient(ctx, config)
+	var w Writer
+	var err error
+	w.client, err = pinecone.NewClient(pinecone.NewClientParams{
+		ApiKey: config.APIKey,
+	})
 	if err != nil {
-		return nil, fmt.Errorf("failed to create pinecone pineconeClient: %v", err)
+		return nil, fmt.Errorf("error creating Pinecone client: %v", err)
 	}
+	sdk.Logger(ctx).Info().Msg("created pinecone client")
 
-	writer := &Writer{
-		client: pineconeClient,
-		index:  indexConnection,
+	// index urls should have their protocol trimmed
+	host := strings.TrimPrefix(config.Host, "https://")
```

Comment:
What if it starts with http or something completely different?

Reply:
@hariso http should never happen, host is given by pinecone, and there's no selfhosting option. The trimming was written because when the user clicks the `copy host` button it copies the protocol + the host
 
We could document that we want the index host without the protocol in the destination params docs, and error out if the user gives host with protocol. 
No opinion here on what to do

Reply:
> http should never happen

Agreed, especially because Pinecone gives the host, but things might change. In other words, if the code relies on something, then we should validate it as such.